lib/src/main/java/com/diffplug/spotless/java/ImportSorterImpl.java

```diff
@@ -91,7 +90,7 @@ private List<String> sort(List<String> imports, String lineFormat) {
 
 	private ImportSorterImpl(List<String> importOrder, boolean wildcardsLast, boolean semanticSort,
 			Set<String> treatAsPackage, Set<String> treatAsClass) {
-		importsGroups = importOrder.stream().filter(Objects::nonNull).map(order -> new ImportsGroup(order, knownGroupings)).collect(Collectors.toList());
+		importsGroups = importOrder.stream().filter(Objects::nonNull).map(order -> new ImportsGroup(order, knownGroupings)).toList();
 		putStaticItemIfNotExists(importsGroups);
 		putCatchAllGroupIfNotExists(importsGroups);
```

Comment:
I think `Collectors.toList()` returns a mutable array list, which can be added-to in the `putStaticItemIfNotExists` (and the other one). Seems that `.toList()` returns an immutable list which doesn't allow that.

Reply:
Indeed, but the docs of `Collectors.toList()` don't guarantee their mutability, so it was naughty of us to add something to it. The best solution is to use `.collect(toCollection(ArrayList::new))`.